src/Label/Label.js

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import Text from "../Text";
+import CSSModules from "react-css-modules";
+import styles from "./Label.css";
+
+export class Label extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {};
+  }
+
+  render() {
+    const {
+      id,
+      inline,
+      label,
+      leftLabel,
+      required,
+      status,
+      style,
+      tooltip,
+      tooltipPosition,
+      ...others
+    } = this.props;
+
+    const reqText = typeof required === "string" ? required : "*";
+
+    let labelClasses = cx({
+      "labelSpacing": !leftLabel,
+      "label": true
+    });
+
+    let labelContainerClasses = cx({
+      "tooltipRight": tooltipPosition === "right",
+      "inline": leftLabel && inline,
+      "labelContainer": true
+    });
+
+    let labelPadding = cx({
+      "verticalPadding": !leftLabel,
+      "horizontalPadding": true
+    });
+
+    const requiredClasses = cx({
+      "required": true,
+      "required_error": status === "error"
+    });
+
+    return (
+      <div styleName={labelContainerClasses}>
+        <div styleName={cx(labelPadding)}>
+          {label && 
+            <Text
+              {...others}
+              as="label"
+              style={style}
+              styleName={labelClasses}
+              htmlFor={id}
```

Comment:
Feels weird `<Label id={sameId}><TextField id={sameId}>`. Duplicate id pattern normal for labels?